### PR TITLE
create and edit client profile relevant contacts (DEV-169)

### PR DESCRIPTION
**Pull request by Davit-BetterAngels**
DEV-169

### CODE REVIEW

**Comment by shiozakiyuko** [comment]
Hi @Davit-BetterAngels 

2 things I noticed 

1. Can you bring the Remove button next to Reset button as it shows in the mockup?
2. Can you modify the words to "Add more contacts"?
3. Can you bring the "Add more contacts" 8px higher? 

Thanks!

image

**Comment by shiozakiyuko** [comment]
One more thing
Can you align the title of Relevant Contacts and Type of relationship section? Thanks!

image

**Comment by shiozakiyuko** [comment]
I'm sorry, I noticed more... :\

- When the Relevant Contact section gets long, and the dropdown menu shows up, the menu items after "Past Case Manager" (Pet, Sibling, Uncle) are visible if you pull the menu up by forcing but it's not selectable.
- When I mistyped the email address (e.g. not entering the correct email address format) and the phone number (12 digits), the text fields didn't give me the validation message. 
- When I tapped the "Add Relevant Contact" text link multiple times, I produced multiple Type of Relationship dropdown menus. 

image
image
image

**Comment by Davit-BetterAngels** [comment]
@shiozakiyuko 
fields validations are not in AC. I saw a ticket for that. 
remove and px are fixed. 
Multiple relative should work that way as user presses it multiple times. 
Select dropdown is a library issue and not related to this ticket. We need to create a ticket to truck that.

File: libs/expo/betterangels/src/lib/screens/AddEditClient/RelevantContacts/Contact/index.tsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+  const contacts = watch('contacts');
+  const relationship = watch(
+    `contacts[${index}].relationshipToClient` as `contacts.${number}.relationshipToClient`
+  );
+
+  const handleRemove = () => {
+    remove(index);
+    if (contacts?.length === 1) {
+      setValue('contacts', []);
+    }
```
Comment (mikefeldberg) [inline libs/expo/betterangels/src/lib/screens/AddEditClient/RelevantContacts/Contact/index.tsx:37]: Can you explain this to me please? I see that it works but I'm not sure I understand how. If you have one contact, `contacts.length === 1`, when you `remove(index)` doesn't `contacts.length === 0`? I'm probably misunderstanding what operation is being performed on what

File: libs/expo/betterangels/src/lib/screens/AddEditClient/RelevantContacts/Contact/index.tsx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        name={`contacts[${index}].phoneNumber`}
+        control={control}
+      />
+      <Input
+        placeholder="Mailing Address"
+        label="Mailing Address"
+        name={`contacts[${index}].mailingAddress`}
+        control={control}
+      />
+
```
Comment (mikefeldberg) [inline libs/expo/betterangels/src/lib/screens/AddEditClient/RelevantContacts/Contact/index.tsx:121]: missing input for "other" text input. Should map to `relationshipToClientOther`

File: libs/expo/betterangels/src/lib/static/enumDisplayMapping.ts [hunk 3]
```diff
@@ -31,3 +36,20 @@ export const enumDisplayVeteran: { [key in YesNoPreferNotToSayEnum]: string } =
...
+  [RelationshipTypeEnum.CurrentCaseManager]: 'Current Case Manager',
+  [RelationshipTypeEnum.Aunt]: 'Aunt',
+  [RelationshipTypeEnum.Child]: 'Child',
+  [RelationshipTypeEnum.Cousin]: 'Cousin',
+  [RelationshipTypeEnum.Father]: 'Father',
+  [RelationshipTypeEnum.Friend]: 'Friend',
+  [RelationshipTypeEnum.Grandparent]: 'Grandparent',
+  [RelationshipTypeEnum.Mother]: 'Mother',
+  [RelationshipTypeEnum.Organization]: 'Organization',
+  [RelationshipTypeEnum.Other]: 'Other',
```
Comment (mikefeldberg) [inline libs/expo/betterangels/src/lib/static/enumDisplayMapping.ts:50]: move `Other` to bottom of list

File: libs/expo/betterangels/src/lib/screens/AddEditClient/RelevantContacts/index.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  const { control } = useFormContext<
+    UpdateClientProfileInput | CreateClientProfileInput
+  >();
+
+  const { fields, append, remove } = useFieldArray({
+    control,
+    name: 'contacts',
+  });
+
+  const isRelevant = expanded === 'Relevant Contacts';
```
Comment (mikefeldberg) [inline libs/expo/betterangels/src/lib/screens/AddEditClient/RelevantContacts/index.tsx:31]:
```suggestion
  const isRelevantContacts = expanded === 'Relevant Contacts';
```

let's update this everywhere. `isRelevant` is too ambiguous

**Review by mikefeldberg — CHANGES_REQUESTED** [review]

File: libs/expo/betterangels/src/lib/screens/AddEditClient/RelevantContacts/Contact/index.tsx [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+        borderColor: Colors.NEUTRAL_LIGHT,
+        borderWidth: 1,
+        backgroundColor: Colors.WHITE,
+        paddingTop: Spacings.md,
+        paddingBottom: Spacings.lg,
+        paddingHorizontal: Spacings.sm,
+      }}
+    >
+      <TextBold size="lg">{enumDisplayRelevant[relationship]}</TextBold>
+      <Input
```
[the remark this answers is not in the document]
Comment (Davit-BetterAngels) [reply libs/expo/betterangels/src/lib/screens/AddEditClient/RelevantContacts/Contact/index.tsx:94]: added

File: libs/expo/betterangels/src/lib/screens/AddEditClient/RelevantContacts/Contact/index.tsx [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+          title="Remove"
+          onPress={() => handleRemove()}
+          accessibilityHint="Removes Relevant contact"
+        />
+
+        <TextButton
+          color={Colors.PRIMARY}
+          title="Reset"
+          onPress={() => handleReset()}
+          accessibilityHint="Removes Relevant contact"
```
[the remark this answers is not in the document]
Comment (Davit-BetterAngels) [reply libs/expo/betterangels/src/lib/screens/AddEditClient/RelevantContacts/Contact/index.tsx:142]: changed

File: libs/expo/betterangels/src/lib/screens/AddEditClient/RelevantContacts/index.tsx (hunk 4 above)
Comment (Davit-BetterAngels) [reply libs/expo/betterangels/src/lib/screens/AddEditClient/RelevantContacts/index.tsx:31]: updated

File: libs/expo/betterangels/src/lib/screens/AddEditClient/RelevantContacts/Contact/index.tsx (hunk 1 above)
Comment (Davit-BetterAngels) [reply libs/expo/betterangels/src/lib/screens/AddEditClient/RelevantContacts/Contact/index.tsx:37]: don't need this function. previously emptying the state was cousing error. but now it works

File: libs/expo/betterangels/src/lib/screens/AddEditClient/RelevantContacts/Contact/index.tsx [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+      `contacts[${index}].mailingAddress` as `contacts.${number}.mailingAddress`,
+      null
+    );
+  };
+
+  if (!relationship) {
+    return (
+      <Select
+        labelMarginLeft="xs"
+        label="Type of Relationship"
```
[the remark this answers is not in the document]
Comment (Davit-BetterAngels) [reply libs/expo/betterangels/src/lib/screens/AddEditClient/RelevantContacts/Contact/index.tsx:61]: updated select to support bold

File: libs/expo/betterangels/src/lib/static/enumDisplayMapping.ts (hunk 3 above)
Comment (Davit-BetterAngels) [reply libs/expo/betterangels/src/lib/static/enumDisplayMapping.ts:50]: done

File: libs/expo/betterangels/src/lib/screens/AddEditClient/RelevantContacts/Contact/index.tsx (hunk 2 above)
Comment (Davit-BetterAngels) [reply libs/expo/betterangels/src/lib/screens/AddEditClient/RelevantContacts/Contact/index.tsx:121]: updated. Though didn't see in the figma that is why missed

Comment (mikefeldberg) [reply libs/expo/betterangels/src/lib/screens/AddEditClient/RelevantContacts/Contact/index.tsx:129]: can we set this to only display if `contact.relationshipToClient === `RelationshipTypeEnum.Other`?

Comment (Davit-BetterAngels) [reply libs/expo/betterangels/src/lib/screens/AddEditClient/RelevantContacts/Contact/index.tsx:129]: done

**Pull request merged**